src/credentials/tests/TestGroupDataProvider.cpp

```diff
@@ -16,209 +16,937 @@
  *    limitations under the License.
  */
 
-#include <credentials/GroupDataProvider.h>
+#include "TestPersistentStorageDelegate.h"
+#include <credentials/GroupDataProviderImpl.h>
+#include <lib/core/CHIPTLV.h>
 #include <lib/support/CHIPMem.h>
 #include <lib/support/UnitTestRegistration.h>
 #include <nlunit-test.h>
+#include <platform/KeyValueStoreManager.h>
+#include <set>
 #include <string.h>
 
 using namespace chip::Credentials;
+using GroupMapping = GroupDataProvider::GroupMapping;
+using Keyset       = GroupDataProvider::Keyset;
+using GroupState   = GroupDataProvider::GroupState;
+using EpochKey     = GroupDataProvider::EpochKey;
 
 namespace chip {
 namespace app {
 namespace TestGroups {
 
-using KeySet = chip::Credentials::GroupDataProvider::KeySet;
+static const char * kKey1   = "abc/def";
+static const char * kValue1 = "abc/def";
+static const char * kValue2 = "abc/ghi/xyz";
+static const size_t kSize1  = strlen(kValue1) + 1;
+static const size_t kSize2  = strlen(kValue2) + 1;
+
+void TestStorageDelegate(nlTestSuite * apSuite, void * apContext)
+{
+    chip::TestPersistentStorageDelegate delegate;
+
+    char out[128];
+    uint16_t size = static_cast<uint16_t>(sizeof(out));
```

Comment:
It is really bad we need all of this insane casing everywhere due to SyncGetKeyValue using uint16_t for size input type :(

Reply:
Agreed.